packages/components/src/components/SquiggleViewer/ExpressionViewer.tsx

```diff
@@ -42,131 +46,133 @@ const VariableList: React.FC<{
+        children: () => (
+          <>
+            <span className="text-neutral-300">"</span>
+            <span className="text-neutral-600 text-sm">{value.value}</span>
+            <span className="text-neutral-300">"</span>
+          </>
+        ),
+      };
     case "Bool":
-      return (
-        <VariableBox value={value} heading="Boolean">
-          {() => (
-            <span className="text-neutral-600 text-sm font-mono">
-              {value.value.toString()}
-            </span>
-          )}
-        </VariableBox>
-      );
+      return {
+        value: value,
+        heading: "Boolean",
+
+        preview: value.value.toString(),
+        children: () => (
+          <span className="text-neutral-600 text-sm font-mono">
+            {value.value.toString()}
+          </span>
+        ),
+      };
     case "Date":
-      return (
-        <VariableBox value={value} heading="Date">
-          {() => value.value.toDateString()}
-        </VariableBox>
-      );
+      return {
+        value: value,
+        heading: "Date",
+
+        children: () => value.value.toDateString(),
+      };
+
     case "Void":
-      return (
-        <VariableBox value={value} heading="Void">
-          {() => "Void"}
-        </VariableBox>
-      );
+      return {
+        value: value,
+        heading: "Void",
+        children: () => "Void",
+      };
     case "TimeDuration": {
-      return (
-        <VariableBox value={value} heading="Time Duration">
-          {() => <NumberShower precision={3} number={value.value} />}
-        </VariableBox>
-      );
+      return {
+        value: value,
+        heading: "Time Duration",
+
+        children: () => <NumberShower precision={3} number={value.value} />,
+      };
     }
     case "Lambda":
-      return (
-        <VariableBox
-          value={value}
-          heading={`Function(${value.value.parameters().join(", ")})`}
-          renderSettingsMenu={({ onChange }) => {
-            return (
-              <ItemSettingsMenu
-                value={value}
-                onChange={onChange}
-                withFunctionSettings={true}
-              />
-            );
-          }}
-        >
-          {(settings) => {
+      return {
+        value: value,
+        heading: ``,
+        preview: `fn(${value.value.parameters().join(", ")})`,
+        renderSettingsMenu: ({ onChange }) => {
+          return (
+            <ItemSettingsMenu
+              value={value}
+              onChange={onChange}
+              withFunctionSettings={true}
+            />
+          );
+        },
+        children: (settings) => {
+          if (value.value.parameters().length === 0) {
+            const fnResult = value.value.call([], environment);
+            if (fnResult.ok && value.path) {
+              const fnValue: SqValue = fnResult.value.withPath(value.path);
+              return getBoxProps(fnValue).children(settings);
+            }
```

Comment:
In this model, we have a bunch of zero argument functions. 
https://squigglehub.org/users/berekuk/models/macrovariables
<img width="1292" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/24bdd7d6-d49b-43e2-9d67-15f24814b707">

They get called on demand.

Reply:
I'd be happy with having a button run the function. Above, I was assuming that it would get run only once the parent was expanded, not sure though.

Reply:
Do you have other suggestions for `withPath`? I really wasn't sure about that.

If it is really nasty to do well, then it's possible that could make this feature too costly.

Reply:
We could support `()` in paths, I guess (and later `(arg1, arg2)` for calculators).

So even if the function returned a record, the path would be a correct Squiggle expression: `foo.fn(5).field.field2`.

But there might be long-term complications if we do this. For example, if such path is focused, but we evaluated the function lazily, that would mean that we'd want to run the function immediately if and only if the value is focused.

And running functions that return nested values is messy; might not play well with webworkers and async runs. (Right now `fn.call` is sync, but it might become async in the future). It would cause weird waterfalls for `fn(5).foo.fn2("blah").bar`.

(Hmm, this makes me think that async runs should be done with React Suspense; I should look into it...)